fcentricprospections.client/src/pages/homepage.tsx

```diff
@@ -0,0 +1,50 @@
+import { useContext, useState } from "react";
+import { ShopIdContext } from "../contexts/dataContext"; // Make sure the path is correct
+import styles from '../App.module.css'; // Adjust your styles as necessary
+import { Link } from "react-router-dom";
+
+export const Homepage = () => {
+    const { shopNames } = useContext(ShopIdContext); // Get shop names from context
+    const [searchTerm, setSearchTerm] = useState(""); // State for search input
+
+    // Filter shop names based on the search term, ensuring at least 3 characters are typed
+    const filteredShopNames = shopNames.filter(shop => {
+        // Check if the search term is at least 3 characters long
+        if (searchTerm.length < 3) return false; // If less than 3 characters, do not include
+        return shop.name.toLowerCase().includes(searchTerm.toLowerCase()); // Otherwise, filter based on name
+    });
+
+    return (
+        <div className={styles.body}>
+            <header className={styles.header}>
+                <input 
+                    className={styles.inputField} 
+                    type="text" 
+                    placeholder="Search..." 
+                    value={searchTerm}
+                    onChange={(e) => setSearchTerm(e.target.value)} // Update state on input change
+                />
+            </header>
```

Comment:
Do not put input in header, move input field to main.

Delete header, define header in Root to keep it common across the app